spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteManifestsSparkAction.java

```diff
@@ -159,34 +169,49 @@ public RewriteManifests.Result execute() {
   }
 
   private RewriteManifests.Result doExecute() {
-    List<ManifestFile> matchingManifests = findMatchingManifests();
+    List<ManifestFile> rewrittenManifests = Lists.newArrayList();
+    List<ManifestFile> addedManifests = Lists.newArrayList();
+
+    Result dataResult = rewriteManifests(ManifestContent.DATA);
+    Iterables.addAll(rewrittenManifests, dataResult.rewrittenManifests());
+    Iterables.addAll(addedManifests, dataResult.addedManifests());
+
+    Result deletesResult = rewriteManifests(ManifestContent.DELETES);
+    Iterables.addAll(rewrittenManifests, deletesResult.rewrittenManifests());
+    Iterables.addAll(addedManifests, deletesResult.addedManifests());
+
+    if (rewrittenManifests.isEmpty() && addedManifests.isEmpty()) {
```

Comment:
Functionally it is fine. It is checking whether calling `rewriteManifests` for both data and delete has returned EMPTY_RESULT. If so, method is retuning empty result. 

But is just checking `rewrittenManifests.isEmpty()` is enough (instead of both)? Because how can we add a new manifest when rewrittenManifests are empty?

Reply:
I can check if both results are `EMPTY_RESULT` and return based on that.
Do you think that will be more readable?

Reply:
I think just `rewrittenManifests.isEmpty()` check is enough here?

Reply:
Changed.